testing/util/lightclient.go

```diff
@@ -28,1049 +37,833 @@ type TestLightClient struct {
-		require.NoError(l.T, err)
+// WithSupermajority Specifies whether the sync committee bits have supermajority or not
+func WithSupermajority() LightClientOption {
+	return func(l *TestLightClient) {
+		l.supermajority = true
 	}
+}
 
-	l.State = state
-	l.AttestedState = attestedState
-	l.AttestedBlock = signedParent
-	l.Block = signedBlock
-	l.Ctx = ctx
-	l.FinalizedBlock = finalizedBlock
+// WithIncreasedAttestedSlot Specifies the number of slots to increase the attested slot by. This does not affect the finalized block's slot if there is any.
+func WithIncreasedAttestedSlot(increaseAttestedSlotBy int) LightClientOption {
+	return func(l *TestLightClient) {
+		l.increaseAttestedSlotBy = increaseAttestedSlotBy
+	}
+}
 
-	return l
+// WithIncreasedFinalizedSlot Specifies the number of slots to increase the finalized slot by. This DOES NOT affect the attested block's slot. That should be handled separately using WithIncreasedAttestedSlot.
+func WithIncreasedFinalizedSlot(increaseFinalizedSlotBy int) LightClientOption {
+	return func(l *TestLightClient) {
+		l.increaseFinalizedSlotBy = increaseFinalizedSlotBy
+	}
 }
 
-func (l *TestLightClient) SetupTestCapellaFinalizedBlockAltair(blinded bool) *TestLightClient {
+func (l *TestLightClient) setupTestAltair() *TestLightClient {
 	ctx := context.Background()
 
-	slot := primitives.Slot(params.BeaconConfig().CapellaForkEpoch * primitives.Epoch(params.BeaconConfig().SlotsPerEpoch)).Add(1)
-
-	attestedState, err := NewBeaconStateCapella()
-	require.NoError(l.T, err)
-	err = attestedState.SetSlot(slot)
-	require.NoError(l.T, err)
-
-	finalizedBlock, err := blocks.NewSignedBeaconBlock(NewBeaconBlockAltair())
-	require.NoError(l.T, err)
-	finalizedBlock.SetSlot(1)
-	finalizedHeader, err := finalizedBlock.Header()
-	require.NoError(l.T, err)
-	finalizedRoot, err := finalizedHeader.Header.HashTreeRoot()
-	require.NoError(l.T, err)
-
-	require.NoError(l.T, attestedState.SetFinalizedCheckpoint(&ethpb.Checkpoint{
-		Epoch: params.BeaconConfig().AltairForkEpoch - 10,
-		Root:  finalizedRoot[:],
-	}))
-
-	parent := NewBeaconBlockCapella()
-	parent.Block.Slot = slot
-
-	signedParent, err := blocks.NewSignedBeaconBlock(parent)
-	require.NoError(l.T, err)
-
-	parentHeader, err := signedParent.Header()
-	require.NoError(l.T, err)
-	attestedHeader := parentHeader.Header
-
-	err = attestedState.SetLatestBlockHeader(attestedHeader)
-	require.NoError(l.T, err)
-	attestedStateRoot, err := attestedState.HashTreeRoot(ctx)
-	require.NoError(l.T, err)
-
-	// get a new signed block so the root is updated with the new state root
-	parent.Block.StateRoot = attestedStateRoot[:]
-	signedParent, err = blocks.NewSignedBeaconBlock(parent)
-	require.NoError(l.T, err)
+	attestedSlot := primitives.Slot(uint64(params.BeaconConfig().AltairForkEpoch) * uint64(params.BeaconConfig().SlotsPerEpoch)).Add(1)
+	if l.increaseAttestedSlotBy > 0 {
+		attestedSlot = attestedSlot.Add(uint64(l.increaseAttestedSlotBy))
```

Comment:
You can avoid this conversion by changing the `increaseAttestedSlotBy` parameter from `int` to `uint64`